cirq/work/observable_measurement.py

```diff
@@ -62,6 +66,60 @@ def _with_parameterized_layers(
     return total_circuit
 
 
+class StoppingCriteria(abc.ABC):
+    """An abstract object that queries a BitstringAccumulator to figure out
+    whether that `meas_spec` is complete."""
+
+    @abc.abstractmethod
+    def more_repetitions(self, accumulator: BitstringAccumulator) -> int:
+        """Return the number of additional repetitions to take.
+
+        StoppingCriteria should be respectful and have some notion of a
+        maximum number of repetitions per chunk.
+        """
+
+
+@json_serializable_dataclass(frozen=True)
+class VarianceStoppingCriteria(StoppingCriteria):
+    """Stop sampling when average variance per term drops below a variance bound."""
+
+    variance_bound: float
+    repetitions_per_chunk: int = 10_000
+
+    def more_repetitions(self, accumulator: BitstringAccumulator) -> int:
+        if len(accumulator.bitstrings) == 0:
+            return self.repetitions_per_chunk
+
+        cov = accumulator.covariance()
+        n_terms = cov.shape[0]
+        ones = np.ones(n_terms)
+        sum_variance = ones @ cov @ ones.T
+        var_of_the_e = sum_variance / len(accumulator.bitstrings)
+        vpt = var_of_the_e / n_terms
```

Comment:
To make sure I'm reading this properly:
1. get the covariance matrix as provided by the accumulator
2. sum over all elements of the matrix
3. compute the average from the sum

Is that correct? If so, I think 2-3 could be replaced with [`np.average`](https://numpy.org/doc/stable/reference/generated/numpy.average.html).

Reply:
yes, I think you're correct. I took this from @ncrubin

Reply:
there's the extra division by len(accumulator.bitstrings) which happens "in between" the sum and the division by terms, which I'd like to keep separate for clarity of intent